packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -745,12 +771,16 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         this.positionYMove = event.nativeEvent.offsetY;
         this.isMouseLeave = false;
 
-        if (this.mouseIsDown && !this.isRightClick) {
-            this.updateSelection();
-        }
-        if (this.mouseIsDown && this.isRightClick) {
-            this.forceUpdate();
+        if (this.mouseIsDown) {
+            if (this.state.isPanning && !(event.shiftKey && event.ctrlKey)) {
+                this.panHorizontally(event);
+            } else if (!this.isRightClick && !event.ctrlKey) {
```

Comment:
It should not matter which keys are pressed, if you start selecting it should keep selecting until mouse is released.

Reply:
Fixed.